src/test/java/com/commercetools/emailprocessor/email/EmailProcessorTest.java

```diff
@@ -0,0 +1,232 @@
+package com.commercetools.emailprocessor.email;
+
+
+import com.commercetools.emailprocessor.model.Statistics;
+import com.commercetools.emailprocessor.model.TenantConfiguration;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import io.sphere.sdk.json.SphereJsonUtils;
+import io.sphere.sdk.queries.PagedQueryResult;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLStreamHandler;
+import java.net.URLStreamHandlerFactory;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+public class EmailProcessorTest {
+
+
+    private EmailProcessor emailProcessor = null;
+    private TenantConfiguration tenantConfiguration = null;
+    private List<CustomObject<JsonNode>> customObjects = null;
+
+    /**
+     * Setup a email processor mock.
+     *
+     * @throws Exception when the mock cannot created
+     */
+    @Before
+    public void setUp() throws Exception {
+        customObjects = new ArrayList<CustomObject<JsonNode>>();
+        emailProcessor = mock(EmailProcessor.class);
+        Mockito.doCallRealMethod().when(emailProcessor).processEmails(Mockito.any(TenantConfiguration.class));
+        tenantConfiguration = new TenantConfiguration();
+        tenantConfiguration.setProjectKey("testproject");
+        //tenantConfiguration.setClient(mockSphereClient());
+    }
+
+    @Test
+    public void shouldProcessEmails() throws Exception {
+        customObjects.add(createCustomObject("1", "pending", Statistics.RESPONSE_CODE_SUCCESS));
+        customObjects.add(createCustomObject("2", "pending", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("3", "error", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("4", "pending", Statistics.RESPONSE_ERROR_PERMANENT));
```

Comment:
why hardcode all these status messages "pending", "error", ... when you can use the `EMAIL_STATUS_PENDING` constants in EmailProcessor?